ckanext/protected_resources/controllers.py

```diff
@@ -0,0 +1,38 @@
+from ckan.lib.base import BaseController, c
+from ckan import model
+import ckan.plugins.toolkit as tk
+
+class ProtectedResourceController(BaseController):
+
+    def unlock(self, dataset_id, resource_id):
+        try:
+            context = {'model': model, 'user': c.user}
+            if not tk.check_access('protected_resource_lock', context, {'resource_id': resource_id}):
+                tk.abort(401, 'You are not authorized to lock resources')
+
+            tk.get_action('protected_resource_unlock')(context, {'resource_id': resource_id})
+        except tk.ObjectNotFound:
+            tk.abort(404, 'Resource object not found')
+        except tk.NotAuthorized:
+            tk.abort(401, 'You are not authorized to protect this resource')
```

Comment:
2/10
Consider using `unlock`/`lock` consistently in client-facing messages.
While `lock` isn't the wording we chose, I think it gets the point across and is what's on the buttons :)